src/clr-angular/progress/spinner/spinner.ts

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2016-2019 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+import { Directive } from '@angular/core';
+
+export const SPINNER_BASE_CLASS = 'spinner';
+
+@Directive({
+  selector: '[clrSpinner]',
```

Comment:
A possible solution @Shijir mentioned is instead of us adding a new public API we could have the directive match on the existing CSS selector. `selector: '.spinner'`. This would fix anyone already using `.spinner` in existing applications without having to introduce a new public API.

Reply:
I think there are a few options we could take with the spinner API
1. We do nothing and document that the app developer should add the `aria-live` and `aria-busy` when appropriate
2. We use a directive to automatically add the aria labels on all spinners with maybe an input to disable aria label announcements
3. We use a optional directive to add aria labels for convenience but not changing the current default behavior
4. We create a entirely new public API clr-spinner that encapsulates all of the above and determine what are sensible defaults

Questions for the API is do we optimize for the common use case and it may be the wrong default in certain uses? Do we trust the app developer to use the appropriate aria attributes on the application side if we do nothing?

Reply:
I'm not sure for the idea of using the CSS selector. This in a way is limiting the developer free choice to use CSS or component in his product. I think that this has to stay optional, also if this is auto-attach to every selector `.spinner` in feature all later changes on this component will be breaking change if for some reason we need to add something new to them.

For the API part: 
 * I think that we must add a documentation section on when to add aria-live/busy - there some cases that we won't be able to handle just by adding a new component.
 * We could create service to announce some best practices in development mode - for example `@Directive({ selector: 'spinner' }) class Spinner { constructor() { /* check to see if element have aria* attributes if not throw console.log */ }` 
 * Maybe a Service to handle all `aria-live` messages won't be a bad idea 

I think that the developer will be nice to know how to improve their code with `aria*` tags. A documentation page will come in handy. Also, think that this could be a great addition to the "design system" guide.

If some of the components could handle this - this ok - still we have to let the developer choose there classes and components, give them the freedom to create more complex components.

Reply:
@bdryanovski can you setup a time for us to talk about this? Sounds like we need to think through this a bit more.

Reply:
Sure